src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -247,32 +247,34 @@ public boolean checkSolidity(Hash hash, boolean milestone, int maxProcessedTrans
         if(fromHash(tangle, hash).isSolid()) {
             return true;
         }
-        Set<Hash> analyzedHashes = new HashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
+        Set<Hash> analyzedHashes = new LinkedHashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
```

Comment:
```suggestion
        LinkedHashSet<Hash> analyzedHashes = new LinkedHashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
```

We usually like to use interfaces in Java, but unfortunately for `LinkedHashSet` there is no proper abstraction :-(
If someone will replace `LinkedHashSet` with `HashSet` then the correctness of the program will break

Reply:
Changed To LinkedHashSet.